### DISCUSSION TITLE
Is there a best practice to delete logs on forceDelete?

### DISCUSSION THREAD

**Discussion by ChrisAdels** [spatie/laravel-activitylog#862] [Q&A] [answered]
Hey, thx for the nice package. I use softDelete and want to delete all records of the current model on forceDelete. What would be the best practice? Do I have to search with eloquent for the logs and delete them or is there something ready to use? I could only found the general clear command. Thanks

**Comment by Gummibeer** [comment, marked as answer]
I would use an observer and clear the relationship on `deleted`.

**Reply by ChrisAdels** [reply]
Thx, for the quick answer. So in an Observer would check for "isForceDeleting" and then I would delete the entities with Activity::where('subject_type',get_class($_exampleModel_))->where(subject_id, $_exampleId_)->delete(); ? 😅 Or what do you mean with "clear the relationship"? I mean if I just remove the id I would be left with lots of unnecessary blank entries, wouldn't I?

**Reply by Gummibeer** [reply]
No, you can use the activities relationship from the models trait.
https://github.com/spatie/laravel-activitylog/blob/bdc44862aaca39ecbd824133b80dbd7c8017ed7f/src/Traits/LogsActivity.php#L80-L83

**Reply by ChrisAdels** [reply]
ah perfect 🤩 sry, I didn't see that - much better solution 🙈 thanks
